src/RequestTrait.php

```diff
@@ -66,9 +66,7 @@ private function initialize($uri = null, $method = null, $body = 'php://memory',
         $this->uri    = $this->createUri($uri);
         $this->stream = $this->getStream($body, 'wb+');
 
-        list($this->headerNames, $headers) = $this->filterHeaders($headers);
-        $this->assertHeaders($headers);
-        $this->headers = $headers;
+        $this->setHeaders($headers);
```

Comment:
This assumes `setHeaders()` is in the class composing the trait; the trait should define an abstract method, to ensure that the functionality can work. (This should likely be done for the other calls on `$this` as well.)